src/shell/layout/tiling/grabs/resize.rs

```diff
@@ -281,39 +293,42 @@ impl ResizeForkGrab {
                     return true;
                 };
 
-                match tree.get_mut(&self.node).unwrap().data_mut() {
+                let node = tree.get_mut(&self.node).unwrap();
+                let parent = node.parent().cloned();
+
+                let child_orientation = match node.data_mut() {
                     Data::Group {
                         sizes, orientation, ..
                     } => {
-                        if sizes[self.left_up_idx] + sizes[self.left_up_idx + 1]
-                            < match orientation {
-                                Orientation::Vertical => 720,
-                                Orientation::Horizontal => 480,
-                            }
-                        {
+                        if !perform_fork_grab_resize(
+                            &mut sizes[..],
+                            self.left_up_idx,
+                            *orientation,
+                            self.accumulated_delta,
+                        ) {
                             return false;
-                        };
-
-                        let old_size = sizes[self.left_up_idx];
-                        sizes[self.left_up_idx] = (old_size
-                            + self.accumulated_delta.round() as i32)
-                            .max(if self.orientation == Orientation::Vertical {
-                                360
-                            } else {
-                                240
-                            });
-                        let diff = old_size - sizes[self.left_up_idx];
-                        let next_size = sizes[self.left_up_idx + 1] + diff;
-                        sizes[self.left_up_idx + 1] =
-                            next_size.max(if self.orientation == Orientation::Vertical {
-                                360
-                            } else {
-                                240
-                            });
-                        let next_diff = next_size - sizes[self.left_up_idx + 1];
-                        sizes[self.left_up_idx] += next_diff;
+                        }
+
+                        *orientation
                     }
                     _ => unreachable!(),
+                };
+
+                if let Some(parent_left_up_idx) = self.parent_left_up_idx {
+                    if let Some(Data::Group {
+                        orientation, sizes, ..
+                    }) = parent.map(|p| tree.get_mut(&p).unwrap().data_mut())
+                    {
+                        if *orientation == child_orientation {
```

Comment:
This can happen, we simply have to walk up the tree in a `while`-loop and select the next `parent` to check for it's orientation, until we potentially get `None` (which will happen if there isn't any edge in the other direction as we hit the workspace/output border).